src/test/java/ua/chernonog/onlinebookstore/service/impl/BookServiceImplTest.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import ua.chernonog.onlinebookstore.dto.request.book.CreateBookRequestDto;
+import ua.chernonog.onlinebookstore.dto.response.book.BookDtoWithoutCategoryIds;
+import ua.chernonog.onlinebookstore.dto.response.book.BookResponseDto;
+import ua.chernonog.onlinebookstore.entity.Book;
+import ua.chernonog.onlinebookstore.mapper.BookMapper;
+import ua.chernonog.onlinebookstore.repository.book.BookRepository;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @Test
+    @DisplayName("Save a new book and return the saved book details")
+    void save_WithValidBook_ShouldReturnSavedBook() {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle(NEW_BOOK_TITLE);
+        requestDto.setAuthor(NEW_BOOK_AUTHOR);
+        requestDto.setIsbn(NEW_BOOK_ISBN);
+        requestDto.setPrice(NEW_BOOK_PRICE);
+        requestDto.setDescription(NEW_BOOK_DESCRIPTION);
+        requestDto.setCoverImage(NEW_BOOK_COVER_IMAGE);
+
+        BookResponseDto responseDto = new BookResponseDto();
+        responseDto.setTitle(requestDto.getTitle());
+        responseDto.setAuthor(requestDto.getAuthor());
+        responseDto.setIsbn(requestDto.getIsbn());
+        responseDto.setPrice(requestDto.getPrice());
+        responseDto.setDescription(requestDto.getDescription());
+        responseDto.setCoverImage(requestDto.getCoverImage());
+
+        Book newBook = new Book();
+
+        Mockito.when(bookMapper.toModel(requestDto)).thenReturn(newBook);
+        Mockito.when(bookRepository.save(newBook)).thenReturn(newBook);
+        Mockito.when(bookMapper.toDto(newBook)).thenReturn(responseDto);
+
+        BookResponseDto savedBook = bookService.save(requestDto);
+
+        assertEquals(NEW_BOOK_TITLE, savedBook.getTitle());
+        assertEquals(NEW_BOOK_COVER_IMAGE, savedBook.getCoverImage());
+    }
+
+    @Test
+    @DisplayName("Retrieve all books with pagination")
+    void getAll_ShouldReturnAllBooks() {
+        Book book = new Book();
+        book.setId(1L);
+        book.setTitle(SET_BOOK_TITLE);
+        List<Book> bookList = new ArrayList<>();
+        bookList.add(book);
+
+        Page<Book> page = new PageImpl<>(bookList);
+
+        BookResponseDto responseDto = new BookResponseDto();
+        responseDto.setTitle(SET_BOOK_TITLE);
+
+        Pageable pageable = PageRequest.of(DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE);
+
+        Mockito.when(bookRepository.findAll(pageable)).thenReturn(page);
+        Mockito.when(bookMapper.toDto(book)).thenReturn(responseDto);
+
+        List<BookResponseDto> books = bookService.getAll(pageable);
+
+        assertEquals(1, books.size());
```

Comment:
```suggestion
        assertEquals(page.getSize(), books.size());
```
And the same behavior in other test with collection.

Reply:
Not resolved